src/services.js

```diff
@@ -267,28 +267,13 @@ export class Services {
     ));
   }
 
-  /**
-   * TODO(#22414): Remove this in favor of the async API.
-   * Returns a service to register callbacks we wish to execute when an
-   * amp-form is submitted. This is the sync version used by amp-form only, all
-   * other extensions should use `formSubmitPromiseForDoc` below.
-   * @param {!Element|!./service/ampdoc-impl.AmpDoc} elementOrAmpDoc
-   * @return {../extensions/amp-form/0.1/form-submit-service.FormSubmitService}
-   */
-  static formSubmitForDoc(elementOrAmpDoc) {
-    return /** @type {!../extensions/amp-form/0.1/form-submit-service.FormSubmitService} */ (getServiceForDoc(
-      elementOrAmpDoc,
-      'form-submit-service'
-    ));
-  }
-
   /**
    * Returns a service to register callbacks we wish to execute when an
    * amp-form is submitted.
    * @param {!Element|!./service/ampdoc-impl.AmpDoc} elementOrAmpDoc
    * @return {!Promise<../extensions/amp-form/0.1/form-submit-service.FormSubmitService>}
    */
-  static formSubmitPromiseForDoc(elementOrAmpDoc) {
+  static formSubmitForDoc(elementOrAmpDoc) {
```

Comment:
@cvializ PTAL at FormSubmitService usage above.

Reply:
It should be OK as it is above, since the amp-form extension registers the service. But to make it safe from future changes or races I think it would be better the save a reference to the Promise and `then` it when a service reference is needed, like how the amp-analytics `linker-manager.js` does it.